lxc/remote.go

```diff
@@ -126,7 +126,7 @@ func (c *remoteCmd) addServer(config *lxd.Config, server string, addr string, ac
 	} else if addr[0] == '/' {
 		rScheme = "unix"
 	} else {
-		if !shared.PathExists(addr) {
+		if !shared.PathExists(addr) || shared.IsDir(addr) {
```

Comment:
It seems to me like what we really want here is an IsUnixSocket test, because if this is e.g. a unix char device, we can't connect to it then either.